src/System.Net.Sockets.Legacy/tests/FunctionalTests/DualModeSocketTest.cs

```diff
@@ -1007,7 +1006,7 @@ private void Accept_Helper(IPAddress listenOn, IPAddress connectTo)
                 SocketClient client = new SocketClient(serverSocket, connectTo, port);
                 Socket clientSocket = serverSocket.Accept();
                 Assert.True(clientSocket.Connected);
-                Assert.True(clientSocket.DualMode);
+                Assert.True((listenOn != IPAddress.IPv6Any && !listenOn.IsIPv4MappedToIPv6) || clientSocket.DualMode);
```

Comment:
I'll leave that up to @cipop. We can definitely do so, but it would be a pretty invasive change, since each of these helpers (and all of the consuming tests) would have to be split in two.